twisted/twisted
# #5245 #4800 Improve documentation for t.i.protocol.Protocol.

## Scope and purpose

Fixes #5245 
Fixes #4800 

This is more of an exploratory PR to see how to best use the pydoctor referenece markup.

This branch is named  - 5252-t.i.p.Protocol-apidocs ... but should have been named 5245-t.i.p.Protocol-apidocs

It also tries to fix - https://twistedmatrix.com/trac/ticket/4800 in this same PR

It also tried to fix - https://twistedmatrix.com/trac/ticket/6318

It also tries to discourage the usage of inheritance, as a feedback received over IRC from JP.

---------

If I just use `L{transport}` I get this error

```
 /home/runner/work/twisted/twisted/src/twisted/internet/protocol.py:536: ambiguous ref to transport, could be twisted.internet.protocol.BaseProtocol.transport, twisted.internet.protocol.AbstractDatagramProtocol.transport, twisted.internet.protocol.DatagramProtocol.transport
```

How to test
=========

Docs at https://twisted--1507.org.readthedocs.build/en/1507/api/twisted.internet.protocol.Protocol.html

See link for `output can be sent through the BaseProtocol.transport attribute, `

## Contributor Checklist:

* [x] The associated ticket in Trac is here: https://twistedmatrix.com/trac/ticket/5245
* [x] I ran `tox -e lint` to format my patch to meet the [Twisted Coding Standard](https://twistedmatrix.com/documents/current/core/development/policy/coding-standard.html)
* [x] I have created a newsfragment in src/twisted/newsfragments/ (see: [News files](https://twistedmatrix.com/trac/wiki/ReviewProcess#Newsfiles))
* [NA] I have updated the automated tests.
* [x] I have submitted the associated Trac ticket for review by adding the word `review` to the keywords field in Trac, and putting a link to this PR in the comment; it shows up in https://twisted.reviews/ now.
